Action/ConvertPaymentAction.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+namespace Ledjin\Sagepay\Action;
+
+use Payum\Core\Action\ActionInterface;
+use Payum\Core\Bridge\Spl\ArrayObject;
+use Payum\Core\Exception\RequestNotSupportedException;
+use Payum\Core\Model\PaymentInterface;
+use Payum\Core\Request\Convert;
+use Payum\Core\Security\GenericTokenFactoryInterface;
+
+class ConvertPaymentAction implements ActionInterface
+{
+    /**
+     * @var GenericTokenFactoryInterface
+     */
+    protected $tokenFactory;
+
+    /**
+     * @param GenericTokenFactoryInterface $tokenFactory
+     */
+    public function __construct(GenericTokenFactoryInterface $tokenFactory = null)
+    {
+        $this->tokenFactory = $tokenFactory;
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @param Convert $request
+     */
+    public function execute($request)
+    {
+        RequestNotSupportedException::assertSupports($this, $request);
+
+        /** @var PaymentInterface $payment */
+        $payment = $request->getSource();
+
+        $details = ArrayObject::ensureArrayObject($payment->getDetails());
+
+        $details['VendorTxCode'] = $payment->getNumber();
+        $details['Amount'] = $payment->getTotalAmount() / 100;
```

Comment:
it must not be hard coded. It should be like this: https://github.com/Payum/Payum/blob/master/src/Payum/Paypal/ExpressCheckout/Nvp/Action/ConvertPaymentAction.php#L26

Yes I know It is not related to this PR, just noticed it.